### PR TITLE
Datumises Gang Leaders And Gang Members

### PR DESCRIPTION
**Pull request by Mister-Moriarty**
[Gamemodes] [Internal] [Code Quality]

## About the PR:
Migrates gang leader and gang member antagonists onto the new antagonist datum system, alongside some QoL amendments for gangs.

* Gang leaders and gang members have been fully migrated onto the new datum system, including roundstart, inducted, and admin-spawned gang leaders and gang members.
* Earlier-than-anticipated introduction of a subordinate antagonist system in order to facilitate the datumisation of gang members. Admin tools pertaining to the addition and removal of subordinate antagonists have also been implemented.
* Code pertaining to gang antagonist roles, equipment, induction, and other such mechanics is no longer dependant on the game mode being set to gang war, with a few minor exceptions.
* Selecting a gang name and outfit both now use a TGUI popup window.
* Gang leaders may now chose their gang name from the lists of predesigned gang names. This does not replace the "Randomise" function, merely supplements it.
* Gang radio channels now use private radio channel upgrades, as opposed to distinct headsets.
* Significant cleanup of `gangwar.dm`.

## Why's this needed?
Same rationale as #9366.

### CODE REVIEW

File: code/modules/antagonists/gang/gang_member.dm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		src.gang = src.master.current.get_gang()
+		src.gang.members += new_owner
+
+		. = ..()
+
+		src.gang.leader.current?.antagonist_overlay_refresh(TRUE, FALSE)
+		for(var/datum/mind/M in src.gang.members)
+			M.current?.antagonist_overlay_refresh(TRUE, FALSE)
+
+	Del()
```
Comment (ZeWaka) [inline code/modules/antagonists/gang/gang_member.dm:21]: disposing

File: code/modules/antagonists/gang/gang_member.dm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	New(datum/mind/new_owner, do_equip, do_objectives, do_relocate, silent, source, do_pseudo, do_vr, late_setup, master)
+		src.master = master
+		src.gang = src.master.current.get_gang()
+		src.gang.members += new_owner
+
+		. = ..()
+
+		src.gang.leader.current?.antagonist_overlay_refresh(TRUE, FALSE)
+		for(var/datum/mind/M in src.gang.members)
```
Comment (ZeWaka) [inline code/modules/antagonists/gang/gang_member.dm:18]: is the leader not a member?

File: code/modules/antagonists/gang/gang_leader.dm [hunk 3]
```diff
@@ -0,0 +1,125 @@
+/datum/antagonist/gang_leader
```
Comment (ZeWaka) [inline code/modules/antagonists/gang/gang_leader.dm:1]: i feel like this should be a subtype of the gang member datum?

File: code/modules/antagonists/antagonist_helpers.dm [hunk 4]
```diff
@@ -10,6 +10,32 @@
 				. += antag
 				break
 
+/// Returns a list of all gang datums.
+proc/get_all_gangs()
+	var/list/datum/gang/gangs = list()
```
Comment (ZeWaka) [inline code/modules/antagonists/antagonist_helpers.dm:15]: use `.`

File: code/modules/antagonists/gang/gang_leader.dm (hunk 3 above)
Comment (Mister-Moriarty) [reply code/modules/antagonists/gang/gang_leader.dm:1]: Gang leader datums being a subtype of gang member datums would introduce the issue of gang leaders being treated as subordinate antagonists; already I feel that the two datums are sufficiently different to merit distinct types.

File: code/modules/antagonists/gang/gang_member.dm (hunk 2 above)
Comment (Mister-Moriarty) [reply code/modules/antagonists/gang/gang_member.dm:18]: Gang leaders are not added the the `members` list, no. I'll code document `/datum/gang` to clarify this.

**Pull request merged**